### DISCUSSION TITLE
How do you add a wad key/value to the worldspawn entity? (Half-Life)

### DISCUSSION THREAD

**Discussion by robmikh** [Garux/netradiant-custom#222] [Q&A] [answered]
First off: Thanks for the editor, I've been having fun learning it!

I'm targeting Half-Life with my map and I'm having trouble compiling it as-is. I added a copy of the ZHLT tools to the ZHLT folder, as that's where the build variables seem to look. But the compilation fails, as it can't find textures I've used out of halflife.wad. However, if I edit the map file to add the paths to my wad files, compiling the map succeeds. 

For example:
```
// entity 0
{
"classname" "worldspawn"
"wad" "F:\\hlmapping\\Half-Life\\valve\\halflife.wad;F:\\hlmapping\\Half-Life\\valve\\decals.wad;F:\\hlmapping\\Half-Life\\valve\\xeno.wad"
// brush 0
```

I looked around in the editor and couldn't find how to add references to the wad files to my map. I also didn't see a facility to add arbitrary key/values to entities. Am I looking in the wrong place?

**Comment by Garux** [comment, marked as answer]
Idk what's the HL workflow with wads, in Q1 it's done via `wad` key, just like you did.
Arbitrary key/values may be added to entities via `menu.View.Entity inspector`
![image](https://github.com/user-attachments/assets/dac6649c-46e3-4183-9565-69fc85264f13)

**Reply by robmikh** [reply]
Aha! I didn't realize you could use those for adding key value pairs. When I read the tooltip for the "+" buttons beside the text boxes, I thought the textboxes were for selection rather than adding pairs. But after I hit ENTER on the value textbox it was successfully added to the selected entity. Thanks!
